### PR TITLE
Search: Add loading indicator

### PR DESCRIPTION
**Pull request by half0wl**
This adds a loading indicator to the search modal:
cs-2023-09-13-02 47

### CODE REVIEW

File: src/components/LoadingIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import tw from "twin.macro";
```
Comment (FarazPatankar) [inline src/components/LoadingIndicator.tsx:1]: Just realized docs still uses twin. 🙈

File: src/hooks/useDebouncedSearch.ts [hunk 2]
```diff
@@ -104,13 +106,16 @@ export const useDebouncedSearch = <
...
     }
     try {
+      setIsSearching(true);
       const response = await index.search<Response>(query, params);
       trackGoal(FATHOM_SEARCH_PERFORMED_EVT_ID, 0);
       setResults(transformResponse(response));
+      setIsSearching(false);
     } catch (e) {
       console.error(`Search for query "${query}" failed (${e})`);
+      setIsSearching(false);
```
Comment (FarazPatankar) [inline src/hooks/useDebouncedSearch.ts:116]: Can move this and the one on 113 outside the `catch` to only call it once!

**Review by FarazPatankar — APPROVED** [review]

File: src/components/LoadingIndicator.tsx (hunk 1 above)
Comment (half0wl) [reply src/components/LoadingIndicator.tsx:1]: ye this kinda threw me off..

File: src/hooks/useDebouncedSearch.ts (hunk 2 above)
Comment (half0wl) [reply src/hooks/useDebouncedSearch.ts]: 32a4b010b7a3d5b76bec8ff5af8e63db634fead5

Comment (FarazPatankar) [reply src/hooks/useDebouncedSearch.ts]: ty ty, pre-stamped btw.

**Pull request merged**